week08/CountPrimes/src/countprimes/CountPrimes.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package countprimes;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Leetcode: 204. Count Primes
+ * Problem Link: https://leetcode.com/problems/count-primes/
+ * @author souravpalit
+ */
+public class CountPrimes {
+
+    // TC: O(rootn log logn) SC: O(n)
+    public int countPrimes(int n) {
+        int numOfPrimes = 0;
+        boolean [] primeCuts = new boolean [n + 1];
+        
+        for (int i = 2; i <= Math.sqrt(n); i++) {
```

Comment:
Bhaia, this is a bad practice. In every iteration, it calculates `sqrt(N)` which takes `O(logN)` time. You could easily be precalculated.